src/Firebase/AppCheck/VerifyAppCheckTokenResponse.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase\AppCheck;
+
+use JsonSerializable;
+
+/**
+ * @phpstan-import-type DecodedAppCheckTokenShape from DecodedAppCheckToken
+ *
+ * @phpstan-type VerifyAppCheckTokenResponseShape array{
+ *     appId: non-empty-string,
+ *     token: DecodedAppCheckTokenShape,
+ * }
+ */
+final class VerifyAppCheckTokenResponse implements JsonSerializable
+{
+    /**
+     * @param non-empty string $appId
+     */
+    public function __construct(
+        private string $appId,
+        private DecodedAppCheckToken $token,
+    ) {
+        $this->appId = $appId;
+        $this->token = $token;
+    }
```

Comment:
```suggestion
    public function __construct(
        public readonly string $appId,
        public readonly DecodedAppCheckToken $token,
    ) {
    }
```

This would make the `appId()` and `token()` getter functions obsolete. 

When using `private/public <type> $variable` as constructor parameter, you don't need to set them in the constructor body anymore.

When a property is only set in the constructor, we can always add the `readonly` qualifier 💪